airflow/providers/google/cloud/hooks/datafusion.py

```diff
@@ -469,3 +477,93 @@ def stop_pipeline(self, pipeline_name: str, instance_url: str, namespace: str =
         self._check_response_status_and_data(
             response, f"Stopping a pipeline failed with code {response.status}"
         )
+
+
+class DataFusionAsyncHook(GoogleBaseAsyncHook):
+    """Class to get async hook for Datafusion Async"""
+
+    sync_hook_class = DataFusionHook
+    scopes = ["https://www.googleapis.com/auth/cloud-platform"]
+
+    @staticmethod
+    def _base_url(instance_url: str, namespace: str) -> str:
+        return os.path.join(instance_url, "v3", "namespaces", quote(namespace), "apps")
```

Comment:
It is better to use `urllib.parse.urljoin` function instead of os.path.join. Because the second one is used for file path construction

Reply:
good suggestion, thank you :)